### PR TITLE
Improve add-extension with error in case of multiple matches

### PR DESCRIPTION
**Pull request by emmanuelbernard**
I've done two things:

- make sure that if two extensions match the substring in add-extensions, we show the list instead of taking the random first choice
- use a bit of emoji to better expose what extension has been added, was already there or could not be added
- search artifact-id on top of the existing label and name search

```
$ mvn quarkus:add-extensions -Dextensions=secu,postgres,flyway,panache

[INFO] Scanning for projects...
[INFO]
[INFO] --------------------< com.redhat.maven:maven-test >---------------------
[INFO] Building maven-test 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:add-extensions (default-cli) @ maven-test ---
✅ Adding extension io.quarkus:quarkus-elytron-security
✅ Adding extension io.quarkus:quarkus-flyway
✅ Adding extension io.quarkus:quarkus-hibernate-orm-panache
❌ Multiple extensions matching 'postgres'
     * io.quarkus:quarkus-jdbc-postgresql
     * io.quarkus:quarkus-reactive-pg-client
     Be more specific e.g using the exact name or the full gav.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.198 s
[INFO] Finished at: 2019-05-09T09:43:39-04:00
[INFO] ------------------------------------------------------------------------

$ mvn quarkus:add-extensions -Dextensions=secu,jdbc-postgres,flyway,panache
[INFO] Scanning for projects...
[INFO]
[INFO] --------------------< com.redhat.maven:maven-test >---------------------
[INFO] Building maven-test 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:add-extensions (default-cli) @ maven-test ---
👍 Skipping extension io.quarkus:quarkus-elytron-security: already present
👍 Skipping extension io.quarkus:quarkus-flyway: already present
👍 Skipping extension io.quarkus:quarkus-hibernate-orm-panache: already present
✅ Adding extension io.quarkus:quarkus-jdbc-postgresql
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.218 s
[INFO] Finished at: 2019-05-09T09:43:56-04:00
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

File: devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java
Comment (dmlloyd) [inline devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java]: Does this need to use `java.lang.System#lineSeparator`?

**Comment by dmlloyd** [comment]
Needs a rebase.

File: devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java
Comment (emmanuelbernard) [reply devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java]: Thanks. Done.

**Comment by emmanuelbernard** [comment]
Fixed your comment @dmlloyd and added documentation improvement.

**Comment by gsmet** [comment]
I'm still worried about that. A command that worked in the past might cease to work because an additional extension now pops up, making your commands non reproducible and unpredictable.

I'm really not sure this is an improvement.

**Comment by emmanuelbernard** [comment]
Guillaume. This has always been the case. Unless you give a full gav, a add-extension command is not guaranteed to work in the future. Simply because some can add an extension with the same substring or the same tag.

**Comment by emmanuelbernard** [comment]
In other words, I understand your concerns but the only reproducible way is to pass the full gav. 
In fact, in the past you could have had the wrong extension sneaked on you just became of a dupe match.

**Comment by gsmet** [comment]
Yes, exactly. That’s why I created an issue to fix that as I think it’s not something we should do. 

I’ll be fully back tomorrow, I’ll point you to the issue. 

If I’m outnumbered, I can live with it. But reproducibility is IMHO an important thing now. The rule I gave allowed to pass a shortened version of the gav for ease of use.

**Review by stalep — COMMENTED** [review]
I'm not that worried as @gsmet since any new unexpected conflict is fairly well listed in the output from the command. Usability wise I think this change makes it easier for the user.

**Comment by Dufgui** [comment]
hello, i think it's a little conflicted with my pr
https://github.com/quarkusio/quarkus/pull/2252 
Could you take a look and see if there is a compatible approach or if I must discard my PR.

**Comment by gsmet** [comment]
@stalep even if the output guides you, you end up with the instructions not working anymore. I don't think that's something we want. And consider a script for instance.

@emmanuelbernard so, apart from the rationale about reproducibility, the algo I wanted to use for shortened id here: https://github.com/quarkusio/quarkus/pull/2252#discussion_r279697265 .

**Comment by aloubyansky** [comment]
````
Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.8.1:validate (default) on project quarkus-devtools-common: File '/home/vsts/work/1/s/devtools/common/src/main/java/io/quarkus/cli/commands/AddExtensions.java' has not been previously formatted.  Please format file and commit before running validation!
````

**Comment by gsmet** [comment]
I just force-pushed a rebase + formatting fix.

**Comment by FroMage** [comment]
I think the usability improvement is totally worth it. I also think it's trivial to fix @gsmet's concern about scripts and reproductibility with a `--no-incomplete-match` argument.

**Comment by gsmet** [comment]
I think we will regret it in the future but I look outnumbered. So be it.

I'll wait for CI before merging.

**Review by gsmet — APPROVED** [review]
Tested and seems to work as advertised.

**Pull request merged**
